app/src/main/java/com/example/task/CompletedFragment.kt

```diff
@@ -32,7 +42,13 @@ class CompletedFragment : Fragment() {
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-
+        binding.recyclerViewForFragmentCompleted.layoutManager = LinearLayoutManager(activity)
+        lifecycleScope.launch{
+            TaskViewModel().completedTask.observe(viewLifecycleOwner)
```

Comment:
ViewModel instance is already created on line 26. Use that.

Reply:
Never create ViewModel instances using the constructor invocation syntax. The ViewModel won't survive configuration changes etc, defeating the purpose.